Sources/SwiftDocC/Infrastructure/Workspace/GeneratedDataProvider.swift

```diff
@@ -12,11 +12,12 @@ import Foundation
 import SymbolKit
 
 /// A type that provides documentation bundles that it discovers by traversing the local file system.
-public struct GeneratedDataProvider: DocumentationWorkspaceDataProvider {
+public class GeneratedDataProvider: DocumentationWorkspaceDataProvider {
```

Comment:
Yes. This is driven by needing to know what data to return from `contentsOfURL`. That data depend on if the generated bundle contains one or more modules. However, the argument to this method doesn't indicate one or the other so the number of modules (or its derivate values) need to be stored as private state. The information needed to know if the bundle represents one or more modules isn't known until `bundles(options:)` is called so it can't be computed during initialization. 

Given the choice between making one provider a class or making `bundles(options:)` a mutating func I went with the former because it was a smaller change but I can see the argument for making mutation this a formal part of the provider API. 

In practice, a provider is only asked to provide its bundles once and is only asked to provide content for urls in the returned bundle, so this won't have any impact on concurrency. If a developer did set up a shared GeneratedDataProvider with more than one documentation context, the real issue would be calling `bundles(options:)` twice. 

This class isn't `open`, so making it `final` would only make a difference within Swift-DocC. I don't think we have a case for subclassing this but we don't strictly have a case for not allowing subclassing within Swift-DocC either.